microbench/src/main/java/io/netty/handler/codec/http/HttpStatusValueOfBenchmark.java

```diff
@@ -0,0 +1,215 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http;
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import io.netty.util.internal.SuppressJava6Requirement;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.profile.LinuxPerfNormProfiler;
+import org.openjdk.jmh.profile.ProfilerException;
+import org.openjdk.jmh.profile.ProfilerFactory;
+import org.openjdk.jmh.runner.options.ChainedOptionsBuilder;
+import org.openjdk.jmh.runner.options.ProfilerConfig;
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.SplittableRandom;
+import java.util.concurrent.TimeUnit;
+
+@BenchmarkMode(Mode.Throughput)
+@Warmup(iterations = 10, time = 1)
+@Measurement(iterations = 10, time = 1)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@SuppressJava6Requirement(reason = "suppress")
+public class HttpStatusValueOfBenchmark extends AbstractMicrobenchmark {
+    @Param({"1300", "2600", "5300", "11000", "23000"})
+    private int size;
+    private static final SplittableRandom random = new SplittableRandom();
+    private static final DecimalFormat df = new DecimalFormat("##.##%");
+    private CircularLink head;
+    private static final CircularLink polluteDistributeHead;
+    private static final Map<Integer, CircularLink> sizeMap = new HashMap<Integer, CircularLink>();
+
+    static {
+        sizeMap.put(1300, initCircularLink(1300));
+        sizeMap.put(2600, initCircularLink(2600));
+        sizeMap.put(5300, initCircularLink(5300));
+        sizeMap.put(11000, initCircularLink(11000));
+        sizeMap.put(23000, initCircularLink(23000));
+        polluteDistributeHead = initCircularLink(16000);
+        fillPolluteDistributeData(16000, polluteDistributeHead);
+    }
+
+    @Setup(Level.Invocation)
+    public void setup(Blackhole bh) {
+        // Pollute the branch predictor.
+        CircularLink cl = polluteDistributeHead;
+        do {
+            bh.consume(HttpStatusClass.valueOf(cl.value));
+        } while ((cl = cl.next) != polluteDistributeHead);
```

Comment:
Why do you have used the CircularList linked-list?
It is even more dangerous than using arrays of inputs: this nodes will likely be more scattered on the heap and (very important) introduce a data deps which prevent prefetching of next input data (in short, the CPU cannot speculate the next data(s) to be read.
Said that, I am already happy with the current status of the code...so we can move this on

Reply:
@franz1981 , the reason we used the CircularList linked-list is trying to avoid the 'loop unrolling' you [mentioned](https://github.com/netty/netty/pull/13543#issuecomment-1693394051). I will change the CircularList linked-list to arrays and test it again, then compare the results. I will update the results soon.